apis/v1beta1/gateway_types.go

```diff
@@ -473,16 +542,52 @@ type GatewayAddress struct {
 	Value string `json:"value"`
 }
 
+// GatewayStatusAddress describes an address that is bound to a Gateway.
+type GatewayStatusAddress struct {
+	// Type of the address.
+	//
+	// +optional
+	// +kubebuilder:default=IPAddress
```

Comment:
How about enumerating a few types?  IPv4Address, IPv6Address, DNSName, etc.

Reply:
see: https://github.com/kubernetes-sigs/gateway-api/pull/2171#discussion_r1259073811

Reply:
This is a kubebuilder directive for what the default value should be. This gets picked up when generating the CRD definition and the API server will use this value to default during resource creation when it's not set by the user.

An enumeration doesn't apply here.